### PR TITLE
grpc: add formula for v1.1.2

### PR DESCRIPTION
**Pull request by stanley-cheung**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Fixes grpc/homebrew-grpc#97

### CODE REVIEW

File: Formula/grpc.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class Grpc < Formula
+  desc "Next generation open source RPC library and framework"
+  homepage "http://www.grpc.io/"
+  url "https://github.com/grpc/grpc/archive/v1.1.2.tar.gz"
+  sha256 "373ef66b0919bedef70021ba6798eaefe1cb560a2c94ccd2631d38fc9268f524"
+  head "https://github.com/grpc/grpc.git"
+
+  bottle do
+    root_url "http://storage.googleapis.com/grpc-public/bottles"
+  end
```
Comment (MikeMcQuaid) [inline Formula/grpc.rb]: Remove this, we'll be building our own bottles.

File: Formula/grpc.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on "pkg-config" => :run
+  depends_on "protobuf"
+
+  def install
+    # Clone nanopb into third_party/nanopb IF THE DIRECTORY IS EMPTY.
+    # gRPC v1.0 has nanopb as a submodule of grpc, which requires a clone
+    # of nanopb into the directory. gRPC Master removed the submodule and
+    # always have nanopb files in place, in which case a clone would give
+    # error.
+    # TODO(mxyan): find a better solution
```
Comment (MikeMcQuaid) [inline Formula/grpc.rb]: Create a `resource` and `stage` that into the directory instead.

File: Formula/grpc.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  homepage "http://www.grpc.io/"
+  url "https://github.com/grpc/grpc/archive/v1.1.2.tar.gz"
+  sha256 "373ef66b0919bedef70021ba6798eaefe1cb560a2c94ccd2631d38fc9268f524"
+  head "https://github.com/grpc/grpc.git"
+
+  bottle do
+    root_url "http://storage.googleapis.com/grpc-public/bottles"
+  end
+
+  option "with-plugins", "Also install gRPC protoc plugins."
```
Comment (MikeMcQuaid) [inline Formula/grpc.rb]: Let's remove the option and install these by default.

File: Formula/grpc.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      #!/bin/sh
+      [ "$(ls -A third_party/nanopb)" ] || git clone --branch=nanopb-0.3.5 https://github.com/nanopb/nanopb.git third_party/nanopb
+    EOS
+
+    system "make", "install", "prefix=#{prefix}"
+
+    if build.with? "plugins"
+      system "make", "install-plugins", "prefix=#{prefix}"
+
+      # Link the Objective-C plugin to the name protoc expects.
```
Comment (MikeMcQuaid) [inline Formula/grpc.rb]: Why doesn't `grpc` name this correctly? Can you submit an issue upstream? Thanks!

File: Formula/grpc.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      # TODO: Do this renaming on make install, for all languages.
+      bin.install_symlink bin/"grpc_objective_c_plugin" => "protoc-gen-objcgrpc"
+    end
+  end
+
+  test do
+    system Formula["pkg-config"].bin/"pkg-config", "--libs", "grpc"
```
Comment (MikeMcQuaid) [inline Formula/grpc.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/grpc.rb (hunk 1 above)
Comment (stanley-cheung) [inline Formula/grpc.rb]: Removed

File: Formula/grpc.rb (hunk 3 above)
Comment (stanley-cheung) [inline Formula/grpc.rb]: Removed

File: Formula/grpc.rb (hunk 2 above)
Comment (stanley-cheung) [inline Formula/grpc.rb]: I removed this for now and will try to fix it on our side

File: Formula/grpc.rb (hunk 4 above)
Comment (stanley-cheung) [inline Formula/grpc.rb]: Will fix this on our side

File: Formula/grpc.rb (hunk 5 above)
Comment (stanley-cheung) [inline Formula/grpc.rb]: Done. Thanks for the pointer!

**Comment by dunn** [comment]
I'm having trouble merging this; could you rebase it against the current master?

**Comment by stanley-cheung** [comment]
@dunn Done. Just rebased off the latest master. Thanks.

**Comment by dunn** [comment]
Thanks! Merged as cf4337c63afe4e2949478fa9d211ca53e9589c4a.

**Pull request closed without merging**